src/physics/lsm_driver.f90

```diff
@@ -874,9 +1034,72 @@ subroutine lsm(domain,options,dt)
                                   domain%land_mask,                     &
                                   QSFC,                                 &
                                   QFX,                                  &
-                                  domain%skin_temperature%data_2d)
+                                  domain%skin_temperature%data_2d       &
+                                  ,domain%veg_type                      &
+                                  ,domain%terrain%data_2d               & ! terrain height [m] if ht(i,j)>=lake_min_elev -> lake (in case no lake category is provided, but lake model is selected, we need to not run the simple water as well )
+                                !   ,lake_min_elev                        &
+                                !   ,lakeflag                             &
+                                  )
+            endif
+
+            !___________________ Lake model _____________________
+            ! This lake model (ported from WRF V4.4) is run for the grid cells that are defined as lake in the hi-res input file. 
+            ! It also is advised to supply a lake_depth parameter in the hi-res input, otherwise the default depth of 50m is used (see lakeini above)
+            ! It requires the VEGPARM.TBL landuse category to be one which has a separate lake category (i.e. MODIFIED_IGBP_MODIS_NOAH, USGS-RUC or MODI-RUC).
+            ! For the grid cells that are defined as water, but not as lake (i.e. oceans), the simple water model above will be run.
+            if (options%physics%watersurface==kWATER_LAKE) then    ! WRF's lake model
+
+                ! current_precipitation = (domain%accumulated_precipitation%data_2d-RAINBL)+(domain%precipitation_bucket-rain_bucket)*kPRECIP_BUCKET_SIZE  ! analogous to noah calls
+
+                call lake( &
+                    t_phy=domain%temperature%data_3d                            & !-- t_phy         temperature (K)     !Temprature at the mid points (K)
+                    ,p8w=domain%pressure_interface%data_3d                      & !-- p8w           pressure at full levels (Pa)
```

Comment:
are you sure this should be _interface not just `pressure`? "full levels" sounds like mid-points / mass levels? If it is on interfaces, does it need to have nz+1 levels instead of nz

Reply:
Nope, 8w in WRF-speak appears to be at the interface. Do a 'grep -rnw p8w' in the WRF sourcecode if you have any doubts. "_! Naming convention: 8~at => p8w reads as "p-at-w" (w=full levels)_"
Since it is only used to extract the lowest 2 levels, the upper (nz+1) layer is irrelevant:
```
           PBOT    = p8w(i,2,j)
           PSFC    = P8w(i,1,j) 
```

Reply:
ah, thanks I hadn't ever connected 8w with at w-levels!

Reply:
me neither, just found out! Makes more sense now....